incompatibility/incompatibility-report/node-search.go

```diff
@@ -0,0 +1,36 @@
+package incompatibility
+
+import (
+	"errors"
+	"strconv"
+
+	"gopkg.in/yaml.v3"
+)
+
+func findNode(node *yaml.Node, path ...string) (*yaml.Node, error) {
+	if len(path) == 0 {
+		return node, nil
+	}
+	// Sequence Index
+	if node.Kind == yaml.SequenceNode {
+		ind, err := strconv.Atoi(path[0])
+		if err != nil || ind >= len(node.Content) {
+			return nil, errors.New("invalid index parsed")
+		}
+		return findNode(node.Content[ind], path[1:]...)
+	}
+	//Look for key value
+	for i := 0; i < len(node.Content)-1; i += 2 {
+		key := node.Content[i]
```

Comment:
discussed in person: can we better encapsulate this knowledge of increasing the index by 2 and abstract the structure of the SequenceNode?

Reply:
further changed into mapping